src/esp/scene/SceneNode.h

```diff
@@ -53,10 +72,43 @@ class SceneNode : public MagnumObject,
     new U{*this, std::forward<Args>(args)...};
   }
 
+  // Returns sceneNodeTags of SceneNode
+  SceneNodeTags getSceneNodeTags() const { return sceneNodeTags_; }
+
+  /**
+   * @brief Sets sceneNodeTags_ of SceneNode
+   *
+   * @param sceneNodeTags SceneNodeTags to enumset sceneNodeTags_ to
+   */
+  void setSceneNodeTags(SceneNodeTags sceneNodeTags) {
+    sceneNodeTags_ = sceneNodeTags;
+  }
+
+  /**
+   * @brief Add SceneNodeTag to sceneNodeTags_ of SceneNode
+   *
+   * @param sceneNodeTag SceneNodeTag to add to enumset sceneNodeTags_
+   */
+  void addSceneNodeTag(SceneNodeTag sceneNodeTag) {
+    sceneNodeTags_ |= sceneNodeTag;
+  }
+
+  /**
+   * @brief Remove SceneNodeTag from sceneNodeTags_ of SceneNode
+   *
+   * @param sceneNodeTag SceneNodeTag to remove from enumset sceneNodeTags_
+   */
+  void removeSceneNodeTag(SceneNodeTag sceneNodeTag) {
+    SceneNodeTags toRemove = {sceneNodeTag};
+    sceneNodeTags_ &= ~toRemove;
```

Comment:
Cannot be just like this?? `sceneNodeTags_ &= ~sceneNodeTag;`

Reply:
No, unfortunately not, the types are incompatible after the ~ operator is applied.

Reply:
@mosra : Would you take a look? I believe there is some way to do it.

Reply:
This usually happens due to `CORRADE_ENUMSET_OPERATORS()` not being defined. But I see the macro later down the file, so in this case it's because the function is defined too early, before the compiler even has a chance to know there are the operators :) Solution is to have just a declaration here and have the actual definition after the macro -- either marked as `inline` directly in the header or in the cpp file.

Though probably the least-effort solution would be writing `sceneNodeTags_ &= ~SceneNodeTags{sceneNodeTag};` and not relying on the existence of the operators macro :))

Reply:
@vauduong : Did you read the comment here?

Reply:
Thanks @mosra !